### PR TITLE
Josh Salyer

### PR DESCRIPTION
**Pull request by jsalyer23**
I don't understand why question 4 is not passing. The expected report looks the same as the bin/check results. What am I doing wrong?

### CODE REVIEW

File: test_4_filter_numbers.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 def filter_numbers(arr):
   # Remove the `return None` code below and add your own code.
-  return None
-
+  return arr < 10
```
Comment (halfghaninne) [inline test_4_filter_numbers.py]: I see a couple of issues here. First, on this line you are comparing a List (what Python calls Arrays) against a numerical value. These are two data types that can not be compared.

File: test_4_filter_numbers.py [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
+  filter(filter_numbers, arr)
```
Comment (halfghaninne) [inline test_4_filter_numbers.py]: Second: In this line you are calling the function `filter_numbers` within its own definition.

**Comment by halfghaninne** [comment]
@jsalyer23: 

When I run your code in a REPL, I'm not getting the results I'd expect. `filter` looks like an interesting Python function, but maybe one that will complicate the intent of the assignment. 

The assignment intends to address: If you were to build _your own_ kind of `filter` method, called `filter_numbers` that specifically works for filtering numbers larger than 10 out of an array, how would you do it?

The goal is less about being right or even concise, and more about thinking through the steps that some of these pre-packaged functions take to do their work. 

Please let me know if that helps.

**Comment by halfghaninne** [comment]
If you _would_ like to progress using `filter`, take a look at `lambda` which is Python's means of creating anonymous functions.

I found [this write up](http://www.secnetix.de/olli/Python/lambda_functions.hawk) clarifying.

**Comment by jsalyer23** [comment]
Thank you for the feedback. I think I will try to figure it out without using filter.

**Comment by jsalyer23** [comment]
What if I were to use a for loop to go through the array and then check if each element is less than ten?

**Comment by jsalyer23** [comment]
Are you talking about me in Slack? I resubmitted my answer to this question after I made a for loop to iterate over each element in the array and evaluate if it is less than ten.

**Comment by halfghaninne** [comment]
👍

**Pull request closed without merging**